deps.edn

```diff
@@ -1,7 +1,8 @@
 {:deps {org.clojure/clojure {:mvn/version "1.11.1"}
         org.clojure/core.async {:mvn/version "1.5.648"}
         camel-snake-kebab/camel-snake-kebab {:mvn/version "0.4.3"}
-        cheshire/cheshire {:mvn/version "5.11.0"} }
+        cheshire/cheshire {:mvn/version "5.11.0"}
+        funcool/promesa {:mvn/version "8.0.450"} }
```

Comment:
I can see we using `p/let` for clojure-lsp code actions itself instead of bunch of futures, not sure what you think and the tradeoff considering start using promessa there just for that.
We could also consider using on `process-after-changes` to have a more robust/simple mechanism using a executor service for that as you mentioned